### ISSUE TITLE
Startup/Logging incomplete

### ISSUE THREAD

**Issue by RickyCook** [bitnami/bitnami-docker-postgresql#96]
**Description**

Startup often fails with the message:

```
nami    ERROR Unable to start com.bitnami.postgresql: pg_ctl: could not start server
Examine the log output.
```

Log output is written to `/opt/bitnami/postgresql/logs/postgresql.log`, which would be far more helpful to output to stderr so that it can be read.

**Steps to reproduce the issue:**

This was done on Docker for Mac

1. `mkdir $(pwd)/tmp_bitnami_pg`
2. `docker run --name bitnami_pg -i -v $(pwd)/tmp_bitnami_pg/:/bitnami/postgresql bitnami/postgresql:10.6.0`
3. Wait for PG to be running
4. `docker kill bitnami_pg`
5. `chmod 770 $(pwd)/tmp_bitnami_pg/data`
6. `docker run --name bitnami_pg -i -v $(pwd)/tmp_bitnami_pg/:/bitnami/postgresql bitnami/postgresql:10.6.0`

**Describe the results you received:**

```
» docker run -i -v /Users/rickycook/tmp_bitnami_pg/:/bitnami/postgresql bitnami/postgresql:10.6.0

Welcome to the Bitnami postgresql container
Subscribe to project updates by watching https://github.com/bitnami/bitnami-docker-postgresql
Submit issues and feature requests at https://github.com/bitnami/bitnami-docker-postgresql/issues

nami    INFO  Initializing postgresql
postgre INFO  ==> No injected postgresql.conf file found. Creating default postgresql.conf file...
postgre INFO  ==> No injected pg_hba.conf file found. Creating default pg_hba.conf file...
postgre INFO  ==> Deploying PostgreSQL with persisted data...
postgre INFO  ==> Cleaning stale postmaster.pid file...
postgre INFO  ==> Configuring PostgreSQL...
postgre INFO  ==> Configuring replication parameters...
postgre INFO  ==> Configuring permissions for config files...
postgre INFO
postgre INFO  ########################################################################
postgre INFO   Installation parameters for postgresql:
postgre INFO     Persisted data and properties have been restored.
postgre INFO     Any input specified will not take effect.
postgre INFO   This installation requires no credentials.
postgre INFO  ########################################################################
postgre INFO
nami    INFO  postgresql successfully initialized
INFO  ==> Starting postgresql...
nami    ERROR Unable to start com.bitnami.postgresql: pg_ctl: could not start server
Examine the log output.
```

And in `/opt/bitnami/postgresql/logs/postgresql.log`:
```
2018-11-26 02:35:25.481 GMT [57] FATAL:  data directory "/opt/bitnami/postgresql/data" has group or world access
2018-11-26 02:35:25.481 GMT [57] DETAIL:  Permissions should be u=rwx (0700).
2018-11-26 02:37:04.034 GMT [111] FATAL:  data directory "/opt/bitnami/postgresql/data" has group or world access
2018-11-26 02:37:04.034 GMT [111] DETAIL:  Permissions should be u=rwx (0700).
```

**Describe the results you expected:**

Logs that appeared in`/opt/bitnami/postgresql/logs/postgresql.log` should be output to the terminal, rather than simply saying "Examine the log output"

**Additional information you deem important (e.g. issue happens only occasionally):**

This particular error seems to always happen for me when starting the container on Kubernetes/AWS using an EBS volume. Still investigating, but it looks like the permissions get reset. Possibly this should be automatically managed in startup scripts, but that's kind of a separate issue.

**Version**

- Output of `docker version`:

Docker for Mac, where I reproduced minimally:
```
Client: Docker Engine - Community
 Version:           18.09.0
 API version:       1.39
 Go version:        go1.10.4
 Git commit:        4d60db4
 Built:             Wed Nov  7 00:47:43 2018
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          18.09.0
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.10.4
  Git commit:       4d60db4
  Built:            Wed Nov  7 00:55:00 2018
  OS/Arch:          linux/amd64
  Experimental:     true
```

- Output of `docker info`:

```
Containers: 132
 Running: 20
 Paused: 0
 Stopped: 112
Images: 2821
Server Version: 18.09.0
Storage Driver: overlay2
 Backing Filesystem: extfs
 Supports d_type: true
 Native Overlay Diff: true
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge host ipvlan macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 468a545b9edcd5932818eb9de8e72413e616e86e
runc version: 69663f0bd4b60df09991c08812a60108003fa340
init version: fec3683
Security Options:
 seccomp
  Profile: default
Kernel Version: 4.9.125-linuxkit
Operating System: Docker for Mac
OSType: linux
Architecture: x86_64
CPUs: 4
Total Memory: 1.952GiB
Name: linuxkit-025000000001
ID: NKPQ:F62A:CX2O:MT2M:GGQ7:O37R:DWGK:UTJ6:ACK7:A4NL:2MOC:IBJ2
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 186
 Goroutines: 180
 System Time: 2018-11-26T02:58:25.522823646Z
 EventsListeners: 2
HTTP Proxy: gateway.docker.internal:3128
HTTPS Proxy: gateway.docker.internal:3129
Registry: https://index.docker.io/v1/
Labels:
Experimental: true
Insecure Registries:
 127.0.0.0/8
Live Restore Enabled: false
Product License: Community Engine
```

- Output of `docker-compose version` (if applicable):

N/A

**Additional environment details (AWS, VirtualBox, Docker for MAC, physical, etc.):**
Docker for Mac, but the issue isn't specific to this.

**Comment by marcosbc** [comment]
Hi @RickyCook, the error seems to indicate that the permissions in your data directory are wrong. We see you are executing this command:

    chmod 770 $(pwd)/tmp_bitnami_pg/data

Could you follow the same steps but executing this command instead?

    chmod 700 $(pwd)/tmp_bitnami_pg/data

Also, just curious, where did you get the "770" from?

**Comment by RickyCook** [comment]
the chmod was to purposely cause a startup issue in a simple manner; it has little to do with the actual issue i’m describing.

for some reason, my EBS volume seems to be setting permissions to 770. still tracing the cause of this (it’s non-trivial), but the issue remains that the error should be surfaced through stderr, rather than hidden in a log file

**Comment by marcosbc** [comment]
We're going to refactor our Bitnami PostgreSQL container not to use Nami anymore, but Bash scripts (just like [MySQL](https://github.com/bitnami/bitnami-docker-mysql), [Redis](https://github.com/bitnami/bitnami-docker-redis) and others).

When that is done, it is very probable that the logs will be printed to stdout/stderr, resolving this case. I'm going to leave this open for now.

I cannot give an estimate for when that should happen, but according to our estimations it should happen before EOY (although it might end up being delayed).

**Comment by intelfx** [comment]
Duplicate of #88. @marcosbc — looking forward for nami to be dropped; the initialization logic is totally non-transparent now (is it even open-source?) and not easily customized. This has been a huge problem for us now.

**Comment by stephenmoloney** [comment]
> Hi @intelfx, unfortunately our plans on moving our PostgreSQL container to use Nami logic have been postponed, but it should happen somewhere in Q1 2019.

I think you mean to `not use Nami` instead of `use Nami` ?

**Comment by marcosbc** [comment]
Yep @stephenmoloney, sorry for the typo, the new container will not use Nami but Bash scripts.

**Comment by ybizeul** [comment]
I have the same issue with container restarting with wrong privileges, I have to catch the startup and quickly change `chmod 700 /opt/bitnami/postgresql/data`.
Did you fix that issue @RickyCook ?

**Comment by marcosbc** [comment]
Hi all, just letting you know, in case you were not aware, that the PostgreSQL container was moved to Bash last weekend.

@ybizeul That error seems related to when the PostgreSQL container was moved to use a non-root approach. Did this error happen just recently (with the latest container tags)?

**Comment by ybizeul** [comment]
@marcosbc This is part of awx deployment, their (ansible based) deployment described [here](https://github.com/ansible/awx/blob/devel/installer/roles/kubernetes/tasks/main.yml) used `stable/postgresql` helm chart which seems to be calling `bitnami/postgresql:10.7.0` (updated 5 days ago?)

You I update the chart and force an image download ?

**Comment by javsalgar** [comment]
Hi,

I think that updating the chart to a newer version (making sure that you have a backup first) could potentially fix the issues. My advice would be to deploy a new postgresql chart, check that everything works and then migrate the data.

**Comment by ybizeul** [comment]
I just upgraded to chart 4.2.2 (to stay with 10.x), it seems to be working, I'll update if I have any issue

**Comment by alancivita** [comment]
@ybizeul Hi i have exactly the same issue with awx, could you please explain in detail how did you fix it?
thanks

**Comment by ybizeul** [comment]
If you look at the helm charts, you should see one for bitnami / postgresql. Upgrade it to 4.2.2 and you should be good to go. I don’t remember the specifics, but you can upgrade the chart and keep everything else in place (like persistent data).

**Comment by alancivita** [comment]
Many thanks for the quick response!

**Comment by marcosbc** [comment]
It seems this was fixed, so I'm closing this. Feel free to add a comment in case you find any issues.
